### PR TITLE
Issue 204

### PR DESCRIPTION
**Pull request by icui**
# Description

Please describe the changes/features in this pull request. 

# Issue Number

If there is an issue created for these changes, link it here

# Checklist

Please make sure to check developer documentation on specfem docs. 

[] I ran the code through pre-commit to check style
[] My code passes all the integration tests
[] I have added sufficient unittests to test my changes
[] I have added/updated documentation for the changes I am proposing
[] I have updated CMakeLists to ensure my code builds
[] My code builds across all platforms

### CODE REVIEW

**Comment by Rohit-Kakodkar** [comment]
Merging into `main`. Can you change this to `issue-200`

File: include/mesh/coupled_interfaces/coupled_interfaces.hpp [hunk 1]
```diff
@@ -12,6 +12,7 @@ namespace mesh {
  * @brief Information about coupled interfaces
  *
  */
+template <typename specfem::dimension::type DimensionType>
```
Comment (Rohit-Kakodkar) [inline include/mesh/coupled_interfaces/coupled_interfaces.hpp:15]: Can you add a public constexpr static member variable for dimension to this class? i.e. `constexpr static auto dimension = DimensionType`

File: include/mesh/coupled_interfaces/interface_container.hpp [hunk 2]
```diff
@@ -12,9 +12,14 @@ namespace mesh {
...
+template <specfem::dimension::type DimensionType,
+          specfem::element::medium_tag Medium1,
+          specfem::element::medium_tag Medium2>
+struct interface_container;
+
 template <specfem::element::medium_tag Medium1,
           specfem::element::medium_tag Medium2>
-struct interface_container {
+struct interface_container<specfem::dimension::type::dim2, Medium1, Medium2> {
   constexpr static auto medium1_tag = Medium1; ///< Medium 1 tag
```
Comment (Rohit-Kakodkar) [inline include/mesh/coupled_interfaces/interface_container.hpp:23]: We'll need a `constexpr static auto dimension = specfem::dimension::type::dim2`

**Review by Rohit-Kakodkar — CHANGES_REQUESTED** [review]
Minor changes

File: include/mesh/coupled_interfaces/coupled_interfaces.hpp (hunk 1 above)
Comment (icui) [reply include/mesh/coupled_interfaces/coupled_interfaces.hpp:15]: Will do

**Review by lsawade — APPROVED** [review]
Nothing to add to Rohit's review. LGTM

**Review by Rohit-Kakodkar — APPROVED** [review]
LGTM

**Pull request merged**